packages/containerd/containerd-config-toml_k8s

```diff
@@ -37,3 +37,25 @@ conf_dir = "/etc/cni/net.d"
 endpoint = [{{join_array ", " endpoint }}]
 {{/each}}
 {{/if}}
+
+{{#if settings.container-registry.credentials}}
+{{#each settings.container-registry.credentials}}
+{{#if (eq registry "docker.io" )}}
+[plugins."io.containerd.grpc.v1.cri".registry.configs."registry-1.docker.io".auth]
+{{else}}
+[plugins."io.containerd.grpc.v1.cri".registry.configs."{{registry}}".auth]
+{{/if}}
```

Comment:
Good call out. I tested this and containerd does complain about duplicate entries and then refuse to start:
```
Mar 04 00:27:32 containerd[124031]: containerd: failed to load TOML: /etc/containerd/config.toml: (37, 2): duplicated tables
Mar 04 00:27:32 systemd[1]: containerd.service: Main process exited, code=exited, status=1/FAILURE
Mar 04 00:27:32 systemd[1]: containerd.service: Failed with result 'exit-code'.
Mar 04 00:27:32 systemd[1]: Failed to start containerd container runtime.
```
But I kinda see setting both `docker.io` and `registry-1.docker.io` akin to setting the same registry twice which would also make containerd fail.

We do not enforce uniqueness of the registries in the `container-registry.credentials` settings model. We would essentially need to wrap `Vec<RegistryCredential>` into its own settings model type that would special case `docker.io` and enforce uniqueness of the `registry` field.

Reply:
Alternatively, I can remove this conditional and we can just make users specify `registry-1.docker.io` when they want to set up credentials for dockerhub. And for ECS variants, `https://index.docker.io/v1/`.

Reply:
I'm good with your current approach, after further thought and based on our discussion out of band.

Without the conditional, on `aws-ecs-1` to override registry credentials for Docker Hub, you'd need to specify both so that `host-ctr` and `ecs-agent` both get the form they expect (and ignore the other form). That's awkward and confusing enough that it's worth the extra bit implementation effort to paper over it.